packages/kbn-interpreter/src/server/server_registries.js

```diff
@@ -18,56 +18,35 @@
  */
 
 import { i18n } from '@kbn/i18n';
-import { typesRegistry } from '../common/lib/types_registry';
-import { functionsRegistry as serverFunctions } from '../common/lib/functions_registry';
 import { getPluginPaths } from './get_plugin_paths';
 
-const registries = {
-  serverFunctions: serverFunctions,
-  commonFunctions: serverFunctions,
-  types: typesRegistry,
-};
-
-let resolve = null;
-let called = false;
 
-const populatePromise = new Promise(_resolve => {
-  resolve = _resolve;
-});
+export const populateServerRegistries = registries => {
+  if (!registries) throw new Error('registries are required');
 
-export const getServerRegistries = () => {
-  return populatePromise;
-};
+  return new Promise(resolve => {
+    const remainingTypes = Object.keys(registries);
+    const populatedTypes = {};
 
-export const populateServerRegistries = types => {
-  if (called) {
-    return populatePromise;
-  }
-  called = true;
-  if (!types || !types.length) throw new Error('types is required');
+    const loadType = () => {
+      const type = remainingTypes.pop();
+      getPluginPaths(type).then(paths => {
+        global.canvas = global.canvas || {};
+        global.canvas.register = d => registries[type].register(d);
+        global.canvas.i18n = i18n;
 
-  const remainingTypes = types;
-  const populatedTypes = {};
+        paths.forEach(path => {
+          require(path); // eslint-disable-line import/no-dynamic-require
+        });
 
-  const loadType = () => {
-    const type = remainingTypes.pop();
-    getPluginPaths(type).then(paths => {
-      global.canvas = global.canvas || {};
-      global.canvas.register = d => registries[type].register(d);
-      global.canvas.i18n = i18n;
+        delete global.canvas;
 
-      paths.forEach(path => {
-        require(path); // eslint-disable-line import/no-dynamic-require
+        populatedTypes[type] = registries[type];
+        if (remainingTypes.length) loadType();
```

Comment:
This is curious, why doesn't the linter complain about this? Our [styleguide is pretty clear about requiring curlies](https://github.com/elastic/kibana/blob/master/style_guides/js_style_guide.md#always-use-braces-for-conditionals-and-loops), and I know the tooling was updated so that it worked correctly in Canvas.